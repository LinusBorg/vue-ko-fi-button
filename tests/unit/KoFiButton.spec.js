import { mount } from '@vue/test-utils'
import Button from '@/KoFiButton.vue'

describe('HelloWorld.vue', () => {
  it('renders with default message', () => {
    const wrapper = mount(Button, {
      propsData: {
        username: 'linusborg',
      },
    })
    expect(wrapper.find('span').text()).toMatch('Support Me on Ko-fi')
    expect(wrapper.find('a').attributes('href')).toMatch(
      'https://ko-fi.com/linusborg'
    )
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('renders with custom title', () => {
    const title = 'Gimme my moneys'
    const wrapper = mount(Button, {
      propsData: {
        username: 'linusborg',
        title,
      },
    })
    expect(wrapper.find('span').text()).toMatch(title)
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('renders with custom color from props', () => {
    const wrapper = mount(Button, {
      propsData: {
        username: 'linusborg',
        color: '#CCC',
      },
    })
    expect(wrapper.find('a').element.style.backgroundColor).toMatch(
      'rgb(204, 204, 204)'
    )
  })
})
