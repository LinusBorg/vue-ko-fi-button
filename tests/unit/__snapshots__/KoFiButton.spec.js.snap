// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue renders with custom title 1`] = `
<div class="kofi-btn-container"><a title="Gimme my moneys" href="https://ko-fi.com/linusborg" target="_blank" rel="nofollow noopener" class="kofi-button" style="background-color: rgb(72, 142, 177);"><span class="kofitext"><img src="https://ko-fi.com/img/cuplogo.svg" class="kofiimg">
      Gimme my moneys
    </span></a></div>
`;

exports[`HelloWorld.vue renders with default message 1`] = `
<div class="kofi-btn-container"><a title="Support Me on Ko-fi" href="https://ko-fi.com/linusborg" target="_blank" rel="nofollow noopener" class="kofi-button" style="background-color: rgb(72, 142, 177);"><span class="kofitext"><img src="https://ko-fi.com/img/cuplogo.svg" class="kofiimg">
      Support Me on Ko-fi
    </span></a></div>
`;
